src/views/connectionUI.ts

```diff
@@ -81,37 +79,24 @@ export class ConnectionUI {
         this._errorOutputChannel.show(true);
     }
 
-    // Helper to let user choose a connection from a picklist
-    // Return the ConnectionInfo for the user's choice
-    public showConnections(showExistingConnections: boolean = true): Promise<IConnectionCredentials> {
-        const self = this;
-        return new Promise<IConnectionCredentials>((resolve, reject) => {
-            let picklist: IConnectionCredentialsQuickPickItem[];
-            if (showExistingConnections) {
-                picklist = self._connectionStore.getPickListItems();
-            } else {
-                picklist = [];
-            }
-            if (picklist.length === 0) {
-                // No connections - go to the create profile workflow
-                self.createAndSaveProfile().then(resolvedProfile => {
-                    resolve(resolvedProfile);
-                });
-            } else {
-                // We have recent connections - show them in a picklist
-                self.promptItemChoice({
-                    placeHolder: LocalizedConstants.recentConnectionsPlaceholder,
-                    matchOnDescription: true
-                }, picklist)
-                    .then(selection => {
-                        if (selection) {
-                            resolve(self.handleSelectedConnection(selection));
-                        } else {
-                            resolve(undefined);
-                        }
-                    });
-            }
-        });
+
+    /**
+     * Helper to let user choose a connection from a picklist, or to create a new connection.
+     * Return the ConnectionInfo for the user's choice
+     * @returns The connection picked or created.
+     */
+    public async promptForConnection(): Promise<IConnectionInfo | undefined> {
```

Comment:
it's hard to tell if this new method has the same UX as the previous implementation from looking at the code.  I'm assuming the connection process isn't being changed here?

Reply:
Thanks for looking, yeah I didn't purposely change anything and in my testing it behaves the same as it did previously so there shouldn't be any issues.